### DISCUSSION TITLE
Form.setValues 是否应该添加别名方法来替代参数

### DISCUSSION THREAD

**Discussion by liuweiGL** [alibaba/formily#1429] [Ideas]
## 当前的 API

```ts
form.setValues(values, 'overwrite')

form.setValues(values,  'merge')
```

## 添加别名之后的 API

```ts
form.setValues(values) 

form.mergeValues(values)
```

## 修改方案

1. 在 beta 期间保持 setValues 的行为不变，额外添加 mergeValues 方法。
2. 为 setValues 提供额外的接口签名，使用 @deprecated 标识 setValues(values, strategy) 接口。
3. 正式版可以考虑正式移除 setValues(values, strategy) 接口，仅提供 setValues(value)。

## 出发点

感觉使用别名来替代参数的话，api 的语义更清晰，而且书写也更加方便了（众所皆知程序员是很懒的~）。

**Comment by janryWang** [comment]
你的出发点没任何问题，不过还是不太赞同这种方式，
第一，增加了API记忆成本，统一收敛到setValues上，只需要记住setValues/setInitialValues两个API即可，如果走mergeXXX这种方式的话，就得记住更多的API。
第二，增加API收益并不大，智能提示，目前的方式也是比较友好的，敲一个逗号，就能选择到底是用哪种赋值模式
第三，赋值合并策略这块目前虽然只有复写和合并两种，说不准未来还会有其他合并策略，不适合靠增加API的方式来扩展这些策略

**Reply by janryWang** [reply]
还有一个点就是，不管你是合并还是覆盖，其实都是属于set values这个大语意环境的，这是一个从属关系，而非并列关系，所以我还是倾向于将策略作为参数传递比较合适

**Reply by liuweiGL** [reply]
好的
